log-manager/src/main/java/io/airlift/log/RollingFileMessageOutput.java

```diff
@@ -200,88 +178,21 @@ private RollingFileHandler(String filename, DataSize maxFileSize, DataSize maxTo
         }
     }
 
-    private void start()
-    {
-        thread.start();
-    }
-
-    public synchronized Set<LogFileName> getFiles()
-    {
-        ImmutableSet.Builder<LogFileName> files = ImmutableSet.<LogFileName>builder()
-                .addAll(historyManager.getFiles());
-        if (currentOutputFileName != null) {
-            files.add(currentOutputFileName);
-        }
-        return files.build();
-    }
-
-    @Override
-    public void publish(LogRecord record)
-    {
-        // if closed messages are dropped
-        if (closed.get()) {
-            return;
-        }
-
-        if (!isLoggable(record)) {
-            return;
-        }
-
-        byte[] message;
-        try {
-            message = getFormatter().format(record).getBytes(UTF_8);
-        }
-        catch (Exception e) {
-            // catch any exception to assure logging always works
-            reportError(null, e, FORMAT_FAILURE);
-            return;
-        }
-
-        try {
-            putUninterruptibly(queue, message);
-        }
-        catch (Exception e) {
-            // catch any exception to assure logging always works
-            reportError(null, e, WRITE_FAILURE);
-        }
-
-        // if closed while queueing, try to remove the message just to clean up
-        if (closed.get()) {
-            queue.remove(message);
-        }
-    }
-
     @Override
     public synchronized void flush()
+            throws IOException
     {
         if (currentOutputStream != null) {
-            try {
-                currentOutputStream.flush();
-            }
-            catch (Exception e) {
-                reportError(null, e, FLUSH_FAILURE);
-            }
+            currentOutputStream.flush();
         }
     }
 
     @Override
-    public void close()
+    public synchronized void close()
+            throws IOException
     {
-        closed.set(true);
-
-        putUninterruptibly(queue, POISON_MESSAGE);
-
-        // wait for logging to finish
-        try {
-            thread.join();
-        }
-        catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-        }
-
-        // there shouldn't be any queued methods, but be safe
-        queue.clear();
-
+        currentOutputStream.flush();
+        currentOutputStream.close();
```

Comment:
this should be:
```
 try {
     currentOutputStream.flush();
 }
 finally {
     try {
         currentOutputStream.close();
     }
     finally {
         // wait for compression to finish
         if (compressionExecutor != null) {
             compressionExecutor.shutdown();
             try {
                 compressionExecutor.awaitTermination(1, TimeUnit.MINUTES);
             }
             catch (InterruptedException e) {
                 Thread.currentThread().interrupt();
             }
         }
         currentOutputStream = null;
         currentOutputFile = null;
         currentOutputFileName = null;
         currentFileSize = 0;
     }
 }
```
There might be a more elegant way, but that way I know is safe

Reply:
Actually, use the suppressed exception thing we are doing everywhere else